File: csharp/rotational-cipher/RotationalCipher.cs
using System;
using System.Linq;

public static class RotationalCipher
{
    private static int a { get; } = Convert.ToInt32('a');
    private static int z{ get; } = Convert.ToInt32('z');
    private static int A { get; } = Convert.ToInt32('A');
    private static int Z { get; } = Convert.ToInt32('Z');
    
    public static string Rotate(string text, int shiftKey)
    {
        var crypt = text.Select(c => RotateChar(c, shiftKey));
        return string.Concat(crypt);
    }

    private static char RotateChar(char input, int shiftKey)
    {
        var plain = Convert.ToInt32(input);
        if (a <= plain && plain <= z)
            return Convert.ToChar(a + (plain - a + shiftKey) % 26);
        if (A <= plain && plain <= Z)
            return Convert.ToChar(A + (plain - A + shiftKey) % 26);
        return input;
    }
}

File: fsharp/rotational-cipher/RotationalCipher.fs
module RotationalCipher

open System
open System.Text

let rec rotate shiftKey text =
    let a = int 'a'
    let A = int 'A'
    let rotate_char (c: char): char =
        let ord = int c
        match c with
        | _ when Char.IsUpper c -> char (A + ((ord - A + shiftKey) % 26))
        | _ when Char.IsLower c -> char (a + ((ord - a + shiftKey) % 26))
        | _ -> c
        
    if shiftKey % 26 = 0 then text else text |> String.map rotate_char 
